pkg/cache/cache.go

```diff
@@ -78,22 +78,45 @@ func NewCache(cacheDir string, repoSyncFrequency time.Duration, useGitCaBundle b
 	}
 }
 
-func (c *Cache) OpenRepository(ctx context.Context, repositorySpec *configapi.Repository) (*cachedRepository, error) {
-	ctx, span := tracer.Start(ctx, "Cache::OpenRepository", trace.WithAttributes())
-	defer span.End()
-
+func cacheKey(repositorySpec *configapi.Repository) (string, error) {
 	switch repositoryType := repositorySpec.Spec.Type; repositoryType {
 	case configapi.RepositoryTypeOCI:
 		ociSpec := repositorySpec.Spec.Oci
 		if ociSpec == nil {
-			return nil, fmt.Errorf("oci not configured")
+			return "", fmt.Errorf("oci not configured")
```

Comment:
This code moved from the `OpenRepository()` method that returns with a pointer to the `gteCacheKey()` function that returns with a string. But in case of an error at this point, the original `OpenRepository()` method ultimately will still return with `nil`, as before.